### ISSUE TITLE
Configure AWS to notify us when infra changes.

### ISSUE THREAD

**Issue by glogiotatidis** [mozmeao/infra#1311]
## Current Setup
1. SNS topic 
  - on `us-east-1` 
  - uses `Email`
  - sends to Slack `#meao-notices` via the Slack Email bridge
2. Event Bridge
 - on `us-east-1` (important! IAM events are available only there)
 - Event pattern
```
{
  "source": [
    "aws.iam"
  ],
  "detail-type": [
    "AWS API Call via CloudTrail"
  ],
  "detail": {
    "eventSource": [
      "iam.amazonaws.com"
    ],
    "eventName": [
      {
        "anything-but": "GenerateCredentialReport"
      }
    ]
  }
}
```
 - Targets: SNS topic created in step 1

**Comment by glogiotatidis** [comment]
I don't find IAM in the list of event sources in [CloudWatch](https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#rules:). maybe it's disabled from the parent account? Ideas @mozmeao/theconn ?

/remind me tomorrow

**Comment by metadave** [comment]
hm, that's a bummer. I wonder if they sell this functionality as a different AWS product?

**Comment by duallain** [comment]
If I'm reading the blog linked at the top correctly, it seems like the flow is iam events -> cloudtrail -> cloudwatch -> emails to us.  I think we ought to have access to cloudtrail, but may need to configure 'sending' events from it to cloudwatch.

**Comment by glogiotatidis** [comment]
couldn't find anything on cloudtrail, will research further

**Comment by glogiotatidis** [comment]
@duallain which you get the chance to talk to IT about cloudtrail?

**Comment by duallain** [comment]
@glogiotatidis yes, sorry for not reporting back.

TL;DR is - leave existing cloudtrail as is, provision a new cloudtrail.

IT used to provision cloudtrails via cloudformation, and our account has that setup.  Infosec does monitor the cloudtrail itself, but as long as it doesn't change we shouldn't disrupt them.  It will cost us some $ to make a second cloudtrail. I wasn't able to find a pricing structure to estimate the new cost.

**Comment by glogiotatidis** [comment]
So it seems that we can use CloudWatch for IAM without the need of CloudTrail but only from us-east-1.

I setup a notification for all IAM events sending a ping to #meao-alerts. We can slightly template the message and add / remove information. 

When we agree on the format / channel I'll move this to TF

thoughts @mozmeao/theconn ?

**Comment by metadave** [comment]
I'm ok with #meao-alerts.

Can the message have less detail but a link to the Cloudwatch console?

**Comment by glogiotatidis** [comment]
Status update:

 - I setup a CloudWatch Event Rule and connected that with an SNS topic that in turn sends an email. 
 - We can use message transformations to make the email text look like we want and with the information we want.
 - It turns out that a lot of `GenerateCredentialsReport` events trigger the Rule and increase the noise
 - Researched how we can filter specific events, turns out that CloudWatch Events only supports listing all the events you're interested to. Does not support wildcards or exclusions.
 - But CloudWatch Events is being replaced by EventBridge product which supports some more [advanced filtering](https://docs.aws.amazon.com/eventbridge/latest/userguide/content-filtering-with-event-patterns.html#filtering-prefix-matching) and specifically `anything-but` flag.
 - Had to re-create the SNS topic to `us-east-1` which is the only region from which  CloudWatch can access IAM events.
 - Updated the event pattern to

```json
{
  "source": [
    "aws.iam"
  ],
  "detail-type": [
    "AWS API Call via CloudTrail"
  ],
  "detail": {
    "eventSource": [
      "iam.amazonaws.com"
    ],
    "eventName": [
      {
        "anything-but": "GenerateCredentialReport."
      }
    ]
  }
}
```
 - This appears to be working correctly but I'll keep sending emails to my personal acc for a day or two before I connect it to Slack


FTR the message transformer is the following
```
Input transformer: {"InputPathsMap":{"datetime":"$.detail.eventTime","event_id":"$.detail.eventID","event_type":"$.detail.eventName","username":"$.detail.userIdentity.arn"},"InputTemplate":"\"New AWS IAM event on <datetime> user <username> triggered event <event_id> of type <event_type>. See the event https://console.aws.amazon.com/cloudtrail/home?region=us-east-1#/events?EventId=<event_id> (will become available to CloudTrail in the next 15-30 minutes)\""}
```

**Comment by glogiotatidis** [comment]
OK updated the SNS to point to #meao-notices.

Unfortunately we can't terraform this due to two reasons:
 1. SNS topic uses email and that's not supported by TF because it requires manual verification action
 2. Amazon EventBridge is not support yet.

I'll document what we have in the first comment of this issue for reference.
